eclipse-collections/src/main/java/org/eclipse/collections/impl/collection/mutable/AbstractSynchronizedMutableCollection.java

```diff
@@ -175,4 +181,40 @@ public boolean retainAllIterable(Iterable<?> iterable)
             return this.getDelegate().aggregateBy(groupBy, zeroValueFactory, nonMutatingAggregator);
         }
     }
+
+    @Override
+    public <V> MutableObjectLongMap<V> sumByInt(Function<? super T, ? extends V> groupBy, IntFunction<? super T> function)
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().sumByInt(groupBy, function);
+        }
+    }
+
+    @Override
+    public <V> MutableObjectDoubleMap<V> sumByFloat(Function<? super T, ? extends V> groupBy, FloatFunction<? super T> function)
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().sumByFloat(groupBy, function);
+        }
+    }
+
+    @Override
+    public <V> MutableObjectLongMap<V> sumByLong(Function<? super T, ? extends V> groupBy, LongFunction<? super T> function)
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().sumByLong(groupBy, function);
+        }
+    }
+
+    @Override
+    public <V> MutableObjectDoubleMap<V> sumByDouble(Function<? super T, ? extends V> groupBy, DoubleFunction<? super T> function)
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().sumByDouble(groupBy, function);
+        }
+    }
```

Comment:
These 4 `@Override` annotations are not valid in this class (directly implementing interface).